core/blockdag/blockdag.go

```diff
@@ -1286,13 +1297,26 @@ func (bd *BlockDAG) GetMaturity(target uint, views []uint) uint {
 			queueSet.Add(ib.GetID())
 		}
 	}
-
 	if connected {
-		return maxLayer - targetBlock.GetLayer()
+		matur := maxLayer - targetBlock.GetLayer()
+		result.Store(target, matur)
+		return matur
 	}
 	return 0
 }
 
+// GetMaturity
+func (bd *BlockDAG) BatchGetMaturity(targets map[uint][]uint) (result *sync.Map) {
```

Comment:
Why don't I think closures apply so that the previous interface doesn't change. And multithreading optimization does not belong to BlockDAG itself.
```
// BatchGetMaturity
func (b *BlockChain) BatchGetMaturity(targets map[uint][]uint) (result *sync.Map) {
	result = &sync.Map{}
	wg := sync.WaitGroup{}
	for target, views := range targets {
		wg.Add(1)
		go func(t uint,v []uint) {
			result.LoadOrStore(t, uint(0))
			result.Store(target, b.bd.GetMaturity(t, v))
			wg.Done()
		}(target, views)
	}
	wg.Wait()
	return result
}
```